### DISCUSSION TITLE
Monitor Fargate service from outside Stack?

### DISCUSSION THREAD

**Discussion by iganza** [cdklabs/cdk-monitoring-constructs#306] [Q&A] [answered]
How can I monitor a Fargate service from an outside stack?  I need to import the information, which I am able to do for ex:

FargateService.from_fargate_service_arn(...) call on the CDK can look it up, but this returns an IFargateService, but the monitoring is expecting not that, but 'FargateService' instead.

Is there any way to make this work?

Thanks.

**Comment by ghost** [comment, marked as answer]
The implementation currently relies on getting metrics available from the concrete implementation class rather than the interface.

If the metrics become available in the interface, we can modify it to accept an `IFargateService` (potentially via a change like [this one](https://github.com/aws/aws-cdk/pull/25097)).
